spring-amqp/src/main/java/org/springframework/amqp/support/converter/SerializerMessageConverter.java

```diff
@@ -120,11 +141,40 @@ public Object fromMessage(Message message) throws MessageConversionException {
 		return content;
 	}
 
+	private Object deserialize(ByteArrayInputStream inputStream, Deserializer<Object> deserializer)
+			throws IOException {
+		ClassLoader classLoader = null;
+		try {
+			classLoader = (ClassLoader) new DirectFieldAccessor(deserializer).getPropertyValue("classLoader");
```

Comment:
We have to deal with them injecting a `DefaultDeserializer` so we can't rely on injection (at least in 1.5) but, I agree, we should extract the CL once.